client/web/antrea-ui/src/routes/summary.tsx

```diff
@@ -159,8 +395,16 @@ export default function Summary() {
         // https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
         async function getData() {
             const [controllerInfo, agentInfos, featureGates] = await Promise.all([getControllerInfo(), getAgentInfos(), getFeatureGates()]);
+
+            let finalAgentInfos = agentInfos;
+            // If in development mode, generate fake agent data for UI testing
+            if (import.meta.env.DEV) {
```

Comment:
Even in development mode, one usually connects to an actual K8s cluster and wants real agent data?

I understand that this is convenient, as usually the test clusters used for development are small and don't have a large number of Nodes / agents. So I think we can keep it, but 1) the generation of the synthetic data should be in a separate file if possible for readability, 2) this should be driven by a variable that can be set in `.env.development` / `.env.development.local`. You could set the variable in `.env.development` for reference, but it should be commented out by default. I don't know if there is a better / more standard way of doing it.